core/src/main/java/kafka/server/share/SharePartitionManager.java

```diff
@@ -521,107 +521,107 @@ void maybeProcessFetchQueue() {
             // The queue is already being processed hence avoid re-triggering.
             return;
         }
+        processFetchQueue();
+    }
```

Comment:
I believe that `processFetchQueue()` always releases the `processFetchQueueLock` is the final step when it returns. I would prefer to see the lock released in a try-finally in this `maybeProcessFetchQueue` method to make the code more resilient to change. Having the lock and unlock in separate methods seems unwise, and I believe the code structure makes this quite simple.